<a href="https://colab.research.google.com/github/patimus-prime/ML_notebooks/blob/master/CSID_prelim.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd

In [4]:
!ls

gnomAD_v2.1.1_ENSG00000090402_2023_01_04_01_43_24.csv  sample_data


In [5]:
df = pd.read_csv("./gnomAD_v2.1.1_ENSG00000090402_2023_01_04_01_43_24.csv")

In [7]:
df.head()

Chromosome   Position         rsIDs Reference Alternate          Source  \
0           3  164697082    rs73018867         A         G  gnomAD Genomes   
1           3  164697085   rs139475251         G         T  gnomAD Genomes   
2           3  164697100  rs1366230349         C         G   gnomAD Exomes   
3           3  164697106   rs971273566         T         C   gnomAD Exomes   
4           3  164697109   rs750496465         T         C   gnomAD Exomes   

  Filters - exomes Filters - genomes         Transcript HGVS Consequence  ...  \
0              NaN              PASS  ENST00000264382.3         c.*68T>C  ...   
1              NaN              PASS  ENST00000264382.3         c.*65C>A  ...   
2             PASS               NaN  ENST00000264382.3         c.*50G>C  ...   
3             PASS               NaN  ENST00000264382.3         c.*44A>G  ...   
4             PASS               NaN  ENST00000264382.3         c.*41A>G  ...   

  Homozygote Count European (non-Finnish)  \
0                                     134   
1                                       0   
2                                       0   
3                                       0   
4                                       0   

  Hemizygote Count European (non-Finnish) Allele Count Other  \
0                                       0                165   
1                                       0                  0   
2                                       0                  0   
3                                       0                  0   
4                                       0                  0   

  Allele Number Other  Homozygote Count Other Hemizygote Count Other  \
0                1076                      18                      0   
1                1082                       0                      0   
2                6044                       0                      0   
3                6054                       0                      0   
4                6052                       0                      0   

   Allele Count South Asian  Allele Number South Asian  \
0                         0                          0   
1                         0                          0   
2                         1                      30558   
3                         0                      30572   
4                         0                      30566   

   Homozygote Count South Asian  Hemizygote Count South Asian  
0                             0                             0  
1                             0                             0  
2                             0                             0  
3                             0                             0  
4                             0                             0  

[5 rows x 53 columns]

Variant ID is lost importing to pandas/loading in as CSV, instead of using Hail. But... I'm doing this late and I'm already too far in! So we'll just recreate variant ID, which is Chromosome-Position-Ref-Alt

In [21]:
# create a new column 'D' as the concatenated strings of columns 'A', 'B', and 'C'
df['Variant_ID'] = df['Chromosome'].astype(str) \
 + '-' + df['Position'].astype(str) \
 + '-' + df['Reference'].astype(str) \
 + '-' + df['Alternate'].astype(str)

# move the new column 'D' to the far left by reindexing the columns
df = df.reindex(columns=['Variant_ID'] + df.columns.tolist())

# display the updated dataframe
df.head()

Variant_ID  Chromosome   Position         rsIDs Reference Alternate  \
0  3-164697082-A-G           3  164697082    rs73018867         A         G   
1  3-164697085-G-T           3  164697085   rs139475251         G         T   
2  3-164697100-C-G           3  164697100  rs1366230349         C         G   
3  3-164697106-T-C           3  164697106   rs971273566         T         C   
4  3-164697109-T-C           3  164697109   rs750496465         T         C   

           Source Filters - exomes Filters - genomes         Transcript  ...  \
0  gnomAD Genomes              NaN              PASS  ENST00000264382.3  ...   
1  gnomAD Genomes              NaN              PASS  ENST00000264382.3  ...   
2   gnomAD Exomes             PASS               NaN  ENST00000264382.3  ...   
3   gnomAD Exomes             PASS               NaN  ENST00000264382.3  ...   
4   gnomAD Exomes             PASS               NaN  ENST00000264382.3  ...   

  Hemizygote Count European (non-Finnish) Allele Count Other  \
0                                       0                165   
1                                       0                  0   
2                                       0                  0   
3                                       0                  0   
4                                       0                  0   

  Allele Number Other Homozygote Count Other Hemizygote Count Other  \
0                1076                     18                      0   
1                1082                      0                      0   
2                6044                      0                      0   
3                6054                      0                      0   
4                6052                      0                      0   

   Allele Count South Asian Allele Number South Asian  \
0                         0                         0   
1                         0                         0   
2                         1                     30558   
3                         0                     30572   
4                         0                     30566   

   Homozygote Count South Asian  Hemizygote Count South Asian       Variant_ID  
0                             0                             0  3-164697082-A-G  
1                             0                             0  3-164697085-G-T  
2                             0                             0  3-164697100-C-G  
3                             0                             0  3-164697106-T-C  
4                             0                             0  3-164697109-T-C  

[5 rows x 55 columns]



From gnomAD about:
Stats
v2 release is composed of 125,748 exomes and 15,708 genomes (GRCh37)
gnomAD structural variant (SV) v2.1 represents 10,847 genomes (GRCh37)
v3.1 spans 76,156 genomes (GRCh38)

... Also we get 282832 high quality allele reads in this dataset, max (this is per variant). We have a min of approx. 10k allele reads for one variant with approx 100 reads, so it's only picked up 100 times from a limited dataset, in 10k genomes that pass quality check. I think.



In [22]:
df['ClinVar Clinical Significance'].unique()

array(['Benign', nan, 'Likely benign',
       'Conflicting interpretations of pathogenicity',
       'Uncertain significance', 'Benign/Likely benign',
       'Likely pathogenic', 'Pathogenic/Likely pathogenic', 'Pathogenic'],
      dtype=object)

In [23]:
df['ClinVar Variation ID'].count()


457

457 documented variants in ClinVar -- many of unknown/conflicting significance.

In [74]:
df_path = df[df['ClinVar Clinical Significance'].isin(['Pathogenic',
                                                  'Pathogenic/Likely pathogenic',
                                                  'Likely pathogenic'
                                                  ])]

In [75]:
df_path['ClinVar Variation ID'].count()

18

18 variants, of the 457 recorded in ClinVar, are currently clinically annotated. Note this is for variants caught in ClinVar -- many of which are not annotated. There are many variants not in ClinVar... just starting off small.

If we now however look at the predicted effects for variants from df, we'll see more. VEP looks at genetic conservation, effects on regulatory regions if involved, and ultimately the difference in transcript/protein. It does not dock.

See also:
ClinVar:
https://www.ncbi.nlm.nih.gov/clinvar/?gr=0&term=SI[gene]

dbVar (includes structural vs. clinical):
https://www.ncbi.nlm.nih.gov/dbvar?LinkName=gene_dbvar&from_uid=6476


THIS NEXT ONE GETS ALL THE VARIANTS!

In [56]:
# Let's get full numbers before the following:
print("the total unique variants in df")
df['Variant_ID'].count()

the total unique variants in df


Variant_ID    3470
Variant_ID    3470
dtype: int64

In [31]:
print("the total variants recorded... with clinvar pathogenic annotation")
df_path['Variant_ID'].count()

the total variants recorded... with clinvar pathogenic annotation


Variant_ID    18
Variant_ID    18
dtype: int64

In [26]:
df['VEP Annotation'].unique()

array(['3_prime_UTR_variant', 'synonymous_variant', 'missense_variant',
       'frameshift_variant', 'splice_region_variant', 'intron_variant',
       'inframe_deletion', 'stop_gained', 'splice_donor_variant',
       'splice_acceptor_variant', 'protein_altering_variant',
       'start_lost'], dtype=object)

In [36]:
df_vep = df[df['VEP Annotation'].isin(['frameshift_variant',
                                       'stop_gained',
                                       'splice_donor_variant',
                                       'splice_acceptor_variant',
                                       ])]

In [37]:
df_vep['Variant_ID'].count()

Variant_ID    168
Variant_ID    168
dtype: int64

THAT'S THE BIG NUMBER! 168 variants that are predicted based off VEP. 150 new ones that could lead to some effect, at least based off the limited data in ensembl.

In [55]:
df_vep['Homozygote Count'].sum()

0

In [70]:
df_path['Homozygote Count'].sum()


0

This is really interesting. CSID occurs in patients that are homozygote and neither allele produces functional sucrase. There's also literature that suggests heterozygote individuals also experience symptoms. It may be that no one in gnomAD has CSID, validating prior work on approx. 1/400,000 people having CSID. Just works out they weren't featured in the approx 100,000 genomes.

... But gnomAD also specifically states they try not to include patients with rare diseases, because their specific mutation can potentially be used to identify them/family.

In [64]:
# let's instead look at how many have these variants
df_vep['Allele Count'].sum()

401

So, 200 heterozygotes with one allele predicted bad. 
200/150,000 = 1/750 (0.13%) heterozygote with potential symptoms

Current estimate for homozygote approx. 1/400,000: 
de Leusse, C., et al. "Estimating the prevalence of congenital disaccharidase deficiencies using allele frequencies from gnomAD." Archives de Pédiatrie 29.8 (2022): 599-603.

But yeah, homozygote count is hidden/0 in this genetic database.

---

Most missense are benign for this disease; the ones that are pathogenic or likely pathogenic are above. These could still be modeled. The number is like 102k missense for one "variant" which may as well be the reference. 

In [60]:
df_vep_mis = df[df['VEP Annotation'].isin([
                                       'missense_variant',
                                       ])]

In [61]:
df_vep_mis['Variant_ID'].count()


Variant_ID    1141
Variant_ID    1141
dtype: int64

In [62]:
df_vep_mis['Homozygote Count'].describe()

count      1141.000000
mean        145.155127
std        3426.004690
min           0.000000
25%           0.000000
50%           0.000000
75%           0.000000
max      102305.000000
Name: Homozygote Count, dtype: float64

In [63]:
# this is weird, another day
df_vep_mis['Homozygote Count'].sum()

165622